### PR TITLE
Association

### PR DESCRIPTION
**Pull request by OnlyUnknown**
- Have added new classes [classroom, book, rental]
- Have added a relationship has-to-many between the classroom and the student
- Have added a relationship many-to-many between the person and the book through rental class
- Have fixed litners

### CODE REVIEW

File: Ruby-Project/book.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rental = []
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline Ruby-Project/book.rb]: - [ ] kindly add the method `add_rental` to this class the method should take parameter `person, date` and create a new instance of Rental class

File: Ruby-Project/person_class.rb [hunk 2]
```diff
@@ -9,6 +9,7 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rental = []
```
Comment (VICTOR-CHUKWU) [inline Ruby-Project/person_class.rb:12]: - [ ] kindly add the method `add_rental` to this class the method should take parameter `book, date` and create a new instance of Rental class

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @OnlyUnknown  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 3]
```diff
@@ -1,244 +1,244 @@
...
+
+If you like this project...
+
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+-->
+<!-- ACKNOWLEDGEMENTS -->
+
+
+
```
Comment (laiifuu) [inline README.md]: These two sections are required to be in the `README` file, kindly add them.

File: Ruby-Project/book.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    @title = title
+    @author = author
+    @rental = []
+  end
+
+  def add_rental(date, book)
+    rental = Rental.new(date, book)
+    @rental << rental
+    person.add_rental(rental)
+  end
```
Comment (laiifuu) [inline Ruby-Project/book.rb]: For this method, the only thing that's necessary is to create a new rental. Its creation will automatically add the rental to the rentals list for the Book and Person objects. 
Also the parameters passed should be date and person. For the book we'll be passing `self`. Something like this should do: 
```
def add_rental(person, date)
    Rental.new(date, self, person)
end
```

File: Ruby-Project/classroom.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label
+
```
Comment (laiifuu) [inline Ruby-Project/classroom.rb]: Kindly add an attribute accessor for the `@student` instance variable.

File: Ruby-Project/classroom.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  def initialize(label)
+    @label = label
+    @student = []
+  end
+
+  def add_student(_student)
+    @student.push(Student)
+    Student.classroom = self
```
Comment (laiifuu) [inline Ruby-Project/classroom.rb]: The name for the parameter passed in this method should be the same throughout the entire method: 
```
def add_student(student)
    @student.push(student)
    student.classroom = self
end
```

File: Ruby-Project/person_class.rb [hunk 7]
```diff
@@ -19,6 +20,12 @@ def correct_name
     @name
   end
 
+  def add_rental(date, person)
+    rental = Rental.new(date, person)
+    @rental << rental
+    book.add_rental(rental)
+  end
```
Comment (laiifuu) [inline Ruby-Project/person_class.rb]: As mentionned previously, the only thing that's necessary is to create the rental in this method, something like this should do: 
```
def add_rental(book, date)
    Rental.new(date, book, self)
end
```

File: Ruby-Project/rental.rb [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+  attr_accessor :date
+
+  def initialize(date, person, book)
+    @date = date
+
+    @person = person
+    Person.Rental << self
+
+    @book = book
+    Book.Rental << self
```
Comment (laiifuu) [inline Ruby-Project/rental.rb]: Kindly stick to the way you named your parameters and instance variables: 
```
person.rental << self
book.rental << self
```

File: Ruby-Project/student_class.rb [hunk 9]
```diff
@@ -8,4 +10,9 @@ def initialize(age, classroom, name = 'Unknown', parent_permission: true)
   def play_hooky
     '¯(ツ)/¯'
   end
+
+  def classroom=(classroom)
+    @classroom = classroom
+    classroom.Student.push(self) unless classroom.Student.include?(self)
```
Comment (laiifuu) [inline Ruby-Project/student_class.rb]: Kindly replace `Student` with `student` considering you used a lowercase `s` when naming `@student` in your `Classroom` class.
```
classroom.student.push(self) unless classroom.student.include?(self)
```

**Review by laiifuu — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED :recycle: 

## Hi @OnlyUnknown  :wave: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so-close



 ## Highlights
- [x] Green linters check :heavy_check_mark: 
- [x] Descriptive PR :+1: 

 ## Required Changes ♻️
_Check the comments under the review :point_down: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by frank1738 — APPROVED** [review]
Hi @OnlyUnknown , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Correct use of Gitflow
✅ No linter error
✅ Associations are setup correctly👍 

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**